syntax = "proto3";

package greet;

// The greeting service definition.
service Greeter {
	rpc Unary (HelloRequest) returns (HelloReply);

	rpc ServerStreaming (HelloRequest) returns (stream HelloReply);
	
	rpc ClientStreaming (stream HelloRequest) returns (DelayedReply);

	rpc BothStreaming (stream HelloRequest) returns (stream HelloReply);
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
  string greeting = 2;
}

// The response message containing the greetings.
message HelloReply {
  string message = 1;
}

message DelayedReply {
	string message = 1;
	repeated HelloRequest request = 2;
}